src/ScalarMessaging.js

```diff
@@ -292,22 +292,30 @@ function inviteUser(event, roomId, userId) {
  * arrives) or rejects after a timeout
  *
  * @param {string} widgetId The ID of the widget to wait for
+ * @param {boolean} add True to wait for the widget to be added,
+ *     false to wait for it to be deleted.
  * @returns {Promise} that resolves when the widget is available
  */
-function waitForUserWidget(widgetId) {
+function waitForUserWidget(widgetId, add) {
     return new Promise((resolve, reject) => {
         const currentAccountDataEvent = MatrixClientPeg.get().getAccountData('m.widgets');
-        if (
-            currentAccountDataEvent &&
-            currentAccountDataEvent.getContent() &&
-            currentAccountDataEvent.getContent()[widgetId] !== undefined
-        ) {
+
+        function satisfiesCondition(ev) {
```

Comment:
erm, this might as well be called `function Function(ev)` - at the least can it have a javadoc to say what the condition is used for?

Reply:
Fair: ) Fixed.